src/main/java/duke/task/DeleteTasks.java

```diff
@@ -0,0 +1,110 @@
+package duke.task;
+
+import duke.storage.Storage;
+import duke.command.Command;
+import duke.design.Default;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+
+public class DeleteTasks {
+    /**
+     * Deletes the task(s) according to the task index/indices provided
+     *
+     * @param tasks The array list that stored all the tasks
+     * @param words The array of words that compose the input command
+     */
+    public static void deleteTasks(ArrayList<Task> tasks, String[] words) {
+        switch (words.length) {
+        case 1:
+            Default.showMessage("Sorry, the input task index to delete is missing!");
+            break;
+        case 2:
+            if (Command.isCommandDeleteAll(words[1])) {
+                deleteAllTasks(tasks);
+            } else {
+                deleteSingleTask(tasks, words[1]);
+            }
+            break;
+        default:
+            deleteMultipleTasks(tasks, words);
+        }
+        try {
+            Storage.updateStorage(tasks);
+        } catch (IOException ex) {
+            Default.showMessage("An unexpected error occurs when I try to update the file");
+            System.out.print("\t");
+            ex.printStackTrace();
+        }
+    }
+
+    /**
+     * Deletes all the tasks stored
+     *
+     * @param tasks The array list that stored all the tasks
+     */
+    public static void deleteAllTasks(ArrayList<Task> tasks) {
+        tasks.clear();
+        Default.showMessage("All tasks have been removed! Time to chill?");
+    }
+
+    /**
+     * Deletes the specific task when there is only one task specified
+     *
+     * @param tasks The array list that stored all the tasks
+     * @param word  The array of words that compose the input command
+     */
+    public static void deleteSingleTask(ArrayList<Task> tasks, String word) {
+        try {
+            int index = Integer.parseInt(word);
+            if (tasks.get(index - 1).getDone()) {
+                Default.showMessage("I have removed this task: [" + tasks.get(index - 1).getClassType() + "][X] " + tasks.get(index - 1));
+            } else {
+                Default.showMessage("I have removed this task: [" + tasks.get(index - 1).getClassType() + "][ ] " + tasks.get(index - 1));
+            }
```

Comment:
It looks like code may be repeated here and line 64, perhaps you could [minimise code duplication](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#minimise-code-duplication). You could also build the string first.
```suggestion
            String mark = " ";            
            if (tasks.get(index - 1).getDone()) {
                String mark = "X";
            }
            String message = "I have removed this task: [" + tasks.get(index - 1).getClassType() + 
                    "][" + mark + "] " + tasks.get(index - 1)); // You can still break this up further
            }
```